.circleci/config.yml

```diff
@@ -28,6 +28,7 @@ jobs:
             python3 -m venv venv
             . venv/bin/activate
             pip install .[test]
+            pip install pytest
```

Comment:
should pytest not be installed as part of pip install ? need to somehow define it as a dependency in setup.py.

Side-note: I find setup.py to be quite confusing, I did experiment a bit with https://python-poetry.org/ and liked it better than using setup.py/setuptools. Have not really tested it in depth though

Reply:
agreed on poetry! It looks great. It's not ideal with docker but good enough.